### DISCUSSION TITLE
ipxe memsize too small, initrd command failed

### DISCUSSION THREAD

**Discussion by Feng-zero** [ipxe/ipxe#1125] [Q&A] [answered]
![ipxe](https://github.com/ipxe/ipxe/assets/57554575/15039333-946a-4beb-a370-b2404a2d120d)

I am a beginner in IPxe, please help me.

My initd.img is 100MB, while the ipxe shows memsize as only 64MB. Is it because I need to increase the memory size during compilation or did I execute the command incorrectly? What should I do to manually guide?

**Comment by NiKiZe** [comment]
Did you follow and read the linked page https://ipxe.org/34182006 ?
What hardware is it and how much RAM does it have?

**Reply by Feng-zero** [reply]
I am running the ROM file of the NIC，and the setup interface of the server BIOS showed 256GB of memory. 
I opened #define MEMMAP_ SETTINGS, only 64M displayed on the ipxe interface.
I also enabled the DEBUG=xferbuf compilation option, but there was no output information：
make bin-x86_64-pcbios/MyTest.rom DEBUG=xferbuf
![ipxe1](https://github.com/ipxe/ipxe/assets/57554575/a190ab8d-d0cc-454a-9407-118f1238a843)
initrd.img is 117M

**Reply by NiKiZe** [reply]
How is the initrd loaded? Http? Does the server send content-lenght headers?

Relevant part of the page:
> If you are loading large files you might get this if there isn't enough usable memory. For pcbios boot the memory is especially limited in different ways, even if there is magnitudes more RAM installed in the machine, see http://forum.ipxe.org/showthread.php?tid=11991 and http://lists.ipxe.org/pipermail/ipxe-devel/2016-March/004793.html

**Reply by Feng-zero** [reply]
Both tftp and ftp have reported the same error. I have discussed in the link that there can be 2GB of memory, and my img file is 117MB, which should not exceed the memory size. Is there a problem with my compilation parameters or code configuration? The "memsize" appears to be only 64M, but after downloading "vmlinuz"(8M) using the "kernel" command, the "memsize" becomes 56M.
I will install the HTTP service on the server and see if using HTTP is the same error.

**Reply by NiKiZe** [reply]
Ok, then it is not a http issue. However, you should try a unmodified version of iPXE, you shouldn't really be setting any parameters unless you have a good reason to. (And if you do and have issues, always test with an unmodified version)

**Comment by mcb30** [comment, marked as answer]
@Feng-zero my guess is that you have entered iPXE from the initial "Press Ctrl-B to configure iPXE" prompt that appears during POST.  This will give you a very limited environment since the BIOS has not yet finished some key setup tasks such as creating the system memory map.

Let iPXE boot normally and the problem will go away.

**Reply by Feng-zero** [reply]
Yes! I entered CTRL+B during the BIOS startup process when the Ctrl+B option appeared. Isn't the normal use of IPxe function accessed here? I am a Legacy BIOS system. How can I enter the ipxe boot process normally?

**Reply by Feng-zero** [reply]
Thank you! The problem has been resolved. And after entering ipxe in the later stages, commands can be executed normally.
